azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1599,30 +1599,27 @@ public void run() {
                         executor.getPort(), ConnectorParams.UPDATE_ACTION,
                         null, null, executionIds, updateTimes);
               } catch (final ExecutorManagerException e) {
-                logger.error(e);
+                logger.error("Failed to get update from executor " + executor.getHost(), e);
+                boolean sendUnresponsiveEmail = false;
                 for (final ExecutableFlow flow : entry.getValue()) {
                   final Pair<ExecutionReference, ExecutableFlow> pair =
                       ExecutorManager.this.runningFlows.get(flow.getExecutionId());
 
                   ExecutorManager.this.updaterStage =
```

Comment:
Do we need to check if pair is null?

Reply:
No. As you can see it's never null. Otherwise the existing code would've indeed failed on the next row.

Reply:
The existing code actually checks null on line 1615. If you check other places in the existing code when getting the value from runningFlows, it does the check in most cases. I admit the existing code sometimes has some inconsistent behaviors. So I would prefer not making an assumption here that it would never fail simply based on what the existing code was doing.

Reply:
Ok, this part is complex enough that I haven't tried to figure out if the null check is actually needed or not.

However the existing placement of the null check didn't matter, because in case of pair=null an NPE would be thrown already before the check, so I removed it. **My change doesn't change existing behaviour, it only simplifies code**.

Btw this happens inside a while loop that is protected by try-catch, so if NPE is thrown, it shouldn't happen on the next loop cycle.

That code could be analyzed & improved, but it's not in the scope of this PR. If NPE is ever thrown here, that would be a bug clearly, but fixing it properly might not be as simple as adding a null check. I don't feel comfortable to blindly introduce a new null check at this point, because I'm not sure if that would change the behaviour in some wrong way, but on the other hand it would hide the actual bug so that it can't be noticed as easily as it would when the NPE gets logged as an error.